evalml/automl/engines/engine_base.py

```diff
@@ -0,0 +1,212 @@
+import sys
+import time
+import traceback
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+from evalml.exceptions import PipelineScoreError
+from evalml.model_family import ModelFamily
+from evalml.pipelines import BinaryClassificationPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+from evalml.utils.logger import get_logger, update_pipeline
+
+logger = get_logger(__file__)
+
+
+class EngineBase(ABC):
+    """ Base class for engines, which handles the fitting and evaluation of pipelines."""
+    def __init__(self):
+        """This class represents an "engine" for AutoML, which handles the evaluation of a list of pipelines generated from an AutoML search.
+
+        To use this interface, you must define an `evaluate_batch` method and an `evaluate_pipeline` method.
+        """
+        self.name = "Base Engine"
+        self.X = None
+        self.y = None
+        self.automl = None
+
+    def load_data(self, X, y):
```

Comment:
I had this as a separate function so that we could reuse an engine across multiple searches with different datasets. This is useful for engines like Dask, where this function is overridden and used to distribute the dataset to all workers. 

We could change `__init__` to take in an initial X and y and call `load_data` internally. Thoughts?